src/rdkafka_request.c

```diff
@@ -2359,23 +2382,26 @@ void rd_kafka_ConsumerGroupHeartbeatRequest(
         rd_kafka_buf_write_kstr(rkbuf, rack_id);
         rd_kafka_buf_write_i32(rkbuf, rebalance_timeout_ms);
 
-        if (subscribe_topics) {
+        if (subscribed_topics) {
                 size_t of_TopicsArrayCnt;
-                int topics_cnt = subscribe_topics->cnt;
+                int topics_cnt = subscribed_topics->cnt;
 
                 /* write Topics */
                 of_TopicsArrayCnt = rd_kafka_buf_write_arraycnt_pos(rkbuf);
```

Comment:
we can remove `rd_kafka_buf_write_arraycnt_pos` and `of_TopicsArrayCnt` and use directly `rd_kafka_buf_write_arraycnt`. That is necessary only if the count is to update after some element is skipped.